packages/reactive-element/src/decorators/query-assigned-elements.ts

```diff
@@ -16,11 +16,14 @@ import {decorateProperty} from './base.js';
 import type {ReactiveElement} from '../reactive-element.js';
 import type {QueryAssignedNodesOptions} from './query-assigned-nodes.js';
 
+const NODE_MODE = false;
+const global = NODE_MODE ? globalThis : window;
+
 /**
  * A tiny module scoped polyfill for HTMLSlotElement.assignedElements.
  */
 const slotAssignedElements =
-  window.HTMLSlotElement?.prototype.assignedElements != null
+  global.HTMLSlotElement?.prototype.assignedElements != null
```

Comment:
Is `HTMLSlotElement` going to be defined in Node? Is that in the dom-shim? And what about all the other element classes? Users probably won't use `?.` for these as is done here, right?

Reply:
No, and the broader DOM shim doesn't define `HTMLSlotElement` either. Somebody could be doing e.g. `instanceof HTMLSlotElement` somewhere, but I think we don't need to worry about it for now.